### ISSUE TITLE
Tableau flight sql error with "kamu"."my.trading.holdings" table from example 

### ISSUE THREAD

**Issue by YuriyGavrilov** [kamu-data/kamu-cli#309]
Hi All,

Just receiving this sql error in tableau when trying to select example table "kamu"."my.trading.holdings". 

<img width="1214" alt="Снимок экрана 2023-12-05 в 23 18 01" src="https://github.com/kamu-data/kamu-cli/assets/44679014/6a1bb64f-fb0c-4565-82b0-bce462c878f8">

**To reproduce:**

```
cd trading
kamu init
kamu add . --recursive
kamu pull --all
kamu sql server --flight-sql
```

**error**:

`Error while executing SQL "SELECT "my.trading.holdings"."cum_balance" AS "cum_balance", "my.trading.holdings"."cum_quantity" AS "cum_quantity", "my.trading.holdings"."event_time" AS "event_time", "my.trading.holdings"."offset" AS "offset", "my.trading.holdings"."price" AS "price", "my.trading.holdings"."quantity" AS "quantity", "my.trading.holdings"."settlement" AS "settlement", "my.trading.holdings"."symbol" AS "symbol", "my.trading.holdings"."system_time" AS "system_time" FROM "kamu"."my.trading.holdings" "my.trading.holdings" LIMIT 10000": Error building plan: Schema error: No field named my.trading.holdings.cum_balance. Valid fields are my.trading.holdings.offset, my.trading.holdings.system_time, my.trading.holdings.event_time, my.trading.holdings.symbol, my.trading.holdings.quantity, my.trading.holdings.price, my.trading.holdings.settlement, my.trading.holdings.cum_quantity, my.trading.holdings.cum_balance. SELECT "my.trading.holdings"."cum_balance" AS "cum_balance", "my.trading.holdings"."cum_quantity" AS "cum_quantity", "my.trading.holdings"."event_time" AS "event_time", "my.trading.holdings"."offset" AS "offset", "my.trading.holdings"."price" AS "price", "my.trading.holdings"."quantity" AS "quantity", "my.trading.holdings"."settlement" AS "settlement", "my.trading.holdings"."symbol" AS "symbol", "my.trading.holdings"."system_time" AS "system_time" FROM "kamu"."my.trading.holdings" "my.trading.holdings" LIMIT 10000`

**Context**

- This error only with this dataset my.trading.holdings others works fine.
- Dbeaver select this table (my.trading.holdings) properly 
<img width="1031" alt="Снимок экрана 2023-12-05 в 23 27 36" src="https://github.com/kamu-data/kamu-cli/assets/44679014/473a8f73-ed13-432d-bc22-e959823bbcd0">

**Comment by sergiimk** [comment]
Thanks for the report @YuriyGavrilov.

> Better use Postgres Dialect in Tablau connection to avoid syntax or data type issue

Indeed, SQL queries are executed by Datfusion engine which (on default settings) mimics Postgres dialect, and the error looks like something a difference in quoting styles would cause.

I don't have access to Tableau currently, but [their documentation](https://help.tableau.com/current/pro/desktop/en-us/examples_otherdatabases_jdbc.htm) mentions that there is a dialect selector.

Can you please confirm that the error disappears if you choose the `PostgresSQL` dialect when creating the JDBC source?

Is our [Tableau connectivity](https://docs.kamu.dev/cli/integrations/tableau/) documentation the only thing that needs to be updated here?

**Comment by YuriyGavrilov** [comment]
Yep, it has dialect selector MySQL, Postgres and sql92. I used Postgres in this case I make well calculated field on date and count  rows by dimension. I have trial tableau for next 3 days so can make some additional tests.

**Comment by sergiimk** [comment]
Thanks for confirming @YuriyGavrilov! 

This is likely an upstream issue:
- either Tableau is forming an invalid query
- or Datafusion is not fully conforming to Postgres dialect

Would be nice to know which one is it, but we currently don't have capacity for this. All of us already ran out of Tableau trial periods and Tableau would not provide us a developer licenses to test the compatibility...

Another issue with Tableau is that JDBC sources are not supported in their cloud version, so value of integrating only with desktop is very limited.

We are currently recommending our users to use [Apache Superset](https://superset.apache.org/). It's a bit harder to setup, as it's meant to run on a server not desktop (can run under docker-compose or kubernetes), but provides a far better experience than Tableau+JDBC and can run as multi-tenant service.

I will keep this issue open, but as low priority. If there will be a strong use case for Tableau desktop - will be happy to revisit.

**Comment by YuriyGavrilov** [comment]
@sergiimk Thx for helping, Will try to understand with of the following case it is but same as you mentioned trial has expired. Need to reinstall it, will do it but a bit later.

Regarding Tableau Cloud: each cloud user can install self-hosted server it is for free If cloud license purchased. In S7 we have one dedicated server and a lot of data extracts so this is a good chance to speed up Tableau for us. so mainly this case should help Tableau users which have dedicated server and performance issue with data extracts. Also I think it helps to improve user experience with interactive analysis where response time is important.  

Regarding Superset: I tried to understand it from user side and it seams to be a bit complicated. Mainly for BI Users. Currently we try to power up user experience with Querybook (hope you saw it, https://github.com/pinterest/querybook or https://www.querybook.org). It its fully open sourced by Pinterest and has a good vector to big data, could run with different sql engine, schedule user query, ldap, and could save results on s3 for example. Of source there is a good base for plugins and add-ons. Also I try Redash but the querybook seams to be richer.
